extensions-core/kafka-indexing-service/src/main/java/io/druid/indexing/kafka/KafkaTuningConfig.java

```diff
@@ -44,6 +46,8 @@
   @Deprecated
   private final long handoffConditionTimeout;
   private final boolean resetOffsetAutomatically;
+  private final SinkFactory sinkFactory;
```

Comment:
Please extract an umbrella abstraction like "SegmentStrategy" with createSinkFactory() and createIndexMerger() methods, and inject/serialize/deserialize only it, here and in other places.